neo/Consensus/ConsensusService.cs

```diff
@@ -183,28 +183,7 @@ private void OnChangeViewReceived(ConsensusPayload payload, ChangeView message)
             // additional recovery message response.
             if (!knownHashes.Add(payload.Hash)) return;
             if (message.NewViewNumber <= context.ViewNumber)
-            {
-                if (context.WatchOnly()) return;
-                if (!context.CommitSent())
-                {
-                    bool shouldSendRecovery = false;
-                    // Limit recovery to be sent from, at least, `f` nodes when the request is from a lower view number.
-                    int allowedRecoveryNodeCount = context.F();
-                    for (int i = 0; i < allowedRecoveryNodeCount; i++)
-                    {
-                        var eligibleResponders = context.Validators.Length - 1;
-                        var chosenIndex = (payload.ValidatorIndex + i + message.NewViewNumber) % eligibleResponders;
-                        if (chosenIndex >= payload.ValidatorIndex) chosenIndex++;
-                        if (chosenIndex != context.MyIndex) continue;
-                        shouldSendRecovery = true;
-                        break;
-                    }
-
-                    if (!shouldSendRecovery) return;
-                }
-                Log($"send recovery from view: {message.ViewNumber} to view: {context.ViewNumber}");
-                localNode.Tell(new LocalNode.SendDirectly { Inventory = context.MakeRecoveryMessage() });
-            }
+                OnRecoveryRequestReceived(payload);
```

Comment:
Determining responder should take into account NewViewNumber for better recovery if multiple nodes are dead and one restarts.

Reply:
What do you mean?

Reply:
Let me thing about this again, @jsolman.
Nowdays, we just ask for the next view number (`context.viewnumber + 1`). Maybe the NewViewNumber will not be much useful anymore. I think Erik is right.

I am now thinking about something else! 
Maybe we could use the `hash` of the payload and randomize who respond based on that!
The good thing about that will be that other improvements that were done by including the Timestamp!
What do you think, @erikzhang  and Jeff @jsolman the magister?
Let's merge like this and later improve?

Reply:
Being honest, I am getting a little bit confused with this change.
I am starting to prefer https://github.com/neo-project/neo/pull/707.

PR707 is quite simple. I think we should do that and document it correctly, @igormcoelho.

Why I am afraid:
* This new payload has no protection against DoS as we have in the change view (by filtering payload hash and other mechanisms) that avoids us to process the same payload twice.

The PR 707 is quite simple simple and if we strong comment the code this will be no problem.

Reply:
For this case of receiving `ChangeView` it used to use
`var chosenIndex = (payload.ValidatorIndex + i + message.NewViewNumber) % eligibleResponders;`
Which included taking into account the message.NewViewNumber, so that a node that restarts will get different responders as it counts up its changeView number after a restart. I guess that behavior doesn’t even happen anymore now though, since it won’t count more than one above its current view number until it sees a consensus to change view. So I’ll resolve this issue.